source/utils/types.ts

```diff
@@ -41,10 +41,10 @@ export type ErrorCode =
 	| 'ENETUNREACH'
 	| 'EAI_AGAIN';
 
-export type ResponseType = 'json' | 'buffer' | 'text';
+export type ResponseType = 'json' | 'buffer' | 'text' | '';
```

Comment:
`''` is the default one. It will follow `options.encoding`. For example if `options.encoding` is `null`, it will return a Buffer. If `responseType` is `text`, then that buffer will be converted into String.

Reply:
I'm not sure if I documented that.

Reply:
Kinda weird to use an empty string for that. I think I would prefer either `'auto'`, `'default'`, (or a better name), or `undefined`.

Reply:
Or make the new behavior the default one. I mean let's use `options.encoding` and then convert it into either `json`, `buffer` or `text`. What do you think?

Reply:
But then people couldn't return their own objects, so maybe lets go with `'default'`?

Reply:
> But then people couldn't return their own objects, so maybe lets go with 'default'?

Sure